core/src/main/kotlin/behavior/GlobalApplicationCommandBehavior.kt

```diff
@@ -4,42 +4,46 @@ import dev.kord.common.annotation.KordPreview
 import dev.kord.common.entity.Snowflake
 import dev.kord.core.cache.data.ApplicationCommandData
 import dev.kord.core.entity.Entity
+import dev.kord.core.entity.interaction.ApplicationCommand
 import dev.kord.core.entity.interaction.GlobalApplicationCommand
-import dev.kord.core.entity.interaction.GuildApplicationCommand
-import dev.kord.rest.builder.interaction.GlobalApplicationCommandModifyBuilder
-import dev.kord.rest.builder.interaction.GuildApplicationCommandModifyBuilder
+import dev.kord.rest.builder.interaction.ApplicationCommandModifyBuilder
 import dev.kord.rest.service.InteractionService
 
 @KordPreview
-interface GlobalApplicationCommandBehavior : Entity {
+interface ApplicationCommandBehavior : Entity {
     val applicationId: Snowflake
     val service: InteractionService
-    suspend fun edit(builder: GlobalApplicationCommandModifyBuilder.() -> Unit): GlobalApplicationCommand {
-        val request = GlobalApplicationCommandModifyBuilder().apply(builder).toRequest()
-        val response = service.modifyGlobalApplicationCommand(applicationId, id, request)
-        val data = ApplicationCommandData.from(response)
-        return GlobalApplicationCommand(data, service)
+
+    suspend fun delete()
+}
+
+@KordPreview
+suspend inline fun ApplicationCommandBehavior.edit(builder: ApplicationCommandModifyBuilder.() -> Unit): ApplicationCommand {
+    val request = ApplicationCommandModifyBuilder().apply(builder).toRequest()
+    val response = when (this) { //sealed classes save us
+        is GlobalApplicationCommandBehavior ->
+            service.modifyGlobalApplicationCommand(applicationId, id, request)
+        is GuildApplicationCommandBehavior ->
+            service.modifyGuildApplicationCommand(applicationId, commandId = id, guildId = guildId, request = request)
+        else -> error("unexpected receiver type $this")
     }
+    val data = ApplicationCommandData.from(response)
+    return GlobalApplicationCommand(data, service)
```

Comment:
This should return a guild behavior if the edited commands belong to a command

Reply:
Good catch, I guess we can't put this into a single function then, since the return type won't match completely.